feat: добавлен функционал по добавлению, обновлению и удалению режисс…

Fenris06 opened the pull request:
…ера фильм

fix: в классе FilmDbStorage HashMap заменена на LikedListHashMap
feat: добавлена сортирока фильмов режиссера по году и по дате

Fenris06 committed:
feat: добавлен функционал по добавлению, обновлению и удалению режиссера фильм
fix: в классе FilmDbStorage HashMap заменена на LikedListHashMap
feat: добавлена сортирока фильмов режиссера по году и по дате

Fenris06 committed:
fix: убрал пробел в film

Fenris06 committed:
fix: убрал не используемый импорт в классах

Fenris06 committed:
fix: добавил пробелы в FilmService

Cheshire-Cat74 commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -69,7 +69,12 @@ public List<Film> getPopular(@RequestParam(name = "count", defaultValue = "10",
     @DeleteMapping("/{filmId}")
     public void deleteFilm(@PathVariable Long filmId) {
-        log.debug("received DELETE /films/{filmId} ", filmId);
+        //log.debug("received DELETE /films/{filmId} ", filmId);
```
А зачем тут коммент?

Fenris06 replied:
Идея отказывалась делать коммит ссылаясь на кокой-то аргумент в этом логе. Я так понимаю что из за {filmId} этого.

Cheshire-Cat74 commented on src/main/java/ru/yandex/practicum/filmorate/controller/DirectorController.java:
```diff
@@ -0,0 +1,42 @@
+package ru.yandex.practicum.filmorate.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
```
Я бы развернула импорты

Cheshire-Cat74 commented on src/main/java/ru/yandex/practicum/filmorate/model/Director.java:
```diff
@@ -0,0 +1,16 @@
+package ru.yandex.practicum.filmorate.model;
+
+import lombok.*;
```
Я бы развернула импорты, чтобы лишние не подтягивать

Cheshire-Cat74 commented on src/main/java/ru/yandex/practicum/filmorate/storages/db/FilmDbStorage.java:
```diff
@@ -18,11 +18,7 @@
-import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.util.*;
```
Я бы развернула импорты

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -31,8 +31,13 @@ public class Film {
         genres.add(genre);
     }
+
+    public void addDirector(List<Director> director) {
```
Я бы назвал `addDirectors`

Fenris06 replied:
Исправил

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/storages/FilmStorage.java:
```diff
@@ -19,4 +19,6 @@ public interface FilmStorage {
 
     List<Film> getFilmsRecommendations(long userId);
+
+    List<Film> getSortDirectorFilms(Long directorId, String sort);
```
Предложил бы использовать название `getSortedFilmsByDirector`

Fenris06 replied:
Исправил

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/storages/db/FilmDbStorage.java:
```diff
@@ -190,4 +185,36 @@ public List<Film> getFilmsRecommendations(long userId) {
+                log.debug("Film with director {} sort by {}", directorId, sort);
+                return getFilms(sqlSortDirectorFilmsByLike);
+            default:
+                throw new NotFoundException("This sort type is not supported");
```
Мне кажется больше подходит `ValidationException`

Fenris06 replied:
Исправил

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/storages/DirectorStorage.java:
```diff
@@ -0,0 +1,24 @@
+
+    void delete(Long id);
+
+    void addFilmDirector(Film film);
```
Также кажется будет понятнее если назвать `replaceFilmDirectors` по аналогии с жанрами

Fenris06 replied:
Исправил

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/storages/DirectorStorage.java:
```diff
@@ -0,0 +1,24 @@
+
+    void addFilmDirector(Film film);
+
+    Map<Long, List<Director>> getFilmDirectors(List<Long> filmIds);
```
Может стоит так назвать? `getDirectorsByFilmIds`

Fenris06 replied:
Исправил

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/service/DirectorService.java:
```diff
@@ -0,0 +1,70 @@
+        directorStorage.delete(id);
+    }
+
+    public void addDirectorToBd(Film film) {
```
Можно назвать также как метод в сторадже, после переименования метода в сторадже :)

Fenris06 replied:
Сделал

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/service/DirectorService.java:
```diff
@@ -0,0 +1,70 @@
+        directorStorage.addFilmDirector(film);
+    }
+
+    public List<Film> getListDirectors(List<Film> films) {
```
`getFilmsWithDirectors`

Fenris06 replied:
Исправил

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/service/DirectorService.java:
```diff
@@ -0,0 +1,70 @@
+        return films;
+    }
+
+    public Film getDirector(Film film) {
```
Метод выше делает то же самое. Может использовать только его?

Fenris06 replied:
Поправил

Ivan-Mol commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmService.java:
```diff
@@ -26,18 +27,24 @@ private static void validate(Film film) {
         validate(film);
-        return filmStorage.create(film);
+        filmStorage.create(film);
+        film.setId(film.getId());
```
А что делает эта строчка?

Fenris06 replied:
Она лишняя. Убрал

Ivan-Mol reviewed (changes requested):
Отличная работа, единственное что есть идеи как можно улучшить нэйминг методов)

Fenris06 committed:
fix: поправил названия методов
fix: сделал метод getFilmsWithDirectors общим как для листа фильмов так и для одного фильма

Ivan-Mol reviewed (approved):
Отлично!
